compiler/rustc_smir/src/rustc_smir/context.rs

```diff
@@ -324,7 +324,16 @@ impl<'tcx> Context for TablesWrapper<'tcx> {
     fn instance_ty(&self, def: InstanceDef) -> stable_mir::ty::Ty {
         let mut tables = self.0.borrow_mut();
         let instance = tables.instances[def];
-        instance.ty(tables.tcx, ParamEnv::empty()).stable(&mut *tables)
+        instance.ty(tables.tcx, ParamEnv::reveal_all()).stable(&mut *tables)
```

Comment:
The fact that this uses the empty (reveal-all) param-env seems fraught if you're ever dealing with not-fully-monomorphized functions...

I would `assert!(!instance.has_non_region_param(), "{instance:?} needs further substitution")`.

Reply:
I actually don't think we need the reveal all here. But I like the assertion suggestion

Reply:
`ParamEnv::reveal_all` is correct here for codegen-time function signatures, since it ensures that opaque types are revealed.